host_core/lib/host_core/control_interface/lattice_server.ex

```diff
@@ -88,53 +88,22 @@ defmodule HostCore.ControlInterface.LatticeServer do
 
   ### LINKDEFS
   # These requests are targeted at one host per lattice, changes made as a result
-  # are emitted to the appropriate stream and cached
-
-  # Put a link definition
-  # This will first store the link definition in memory, then publish it to the stream
-  # then publish it directly to the relevant provider via the RPC channel
-  defp handle_request({"linkdefs", "put"}, body, _reply_to, prefix) do
-    Tracer.with_span "Handle Linkdef Put (ctl)", kind: :server do
-      with {:ok, ld} <- Jason.decode(body),
-           true <- has_values(ld, ["actor_id", "contract_id", "link_name", "provider_id"]) do
-        HostCore.Linkdefs.Manager.put_link_definition(
-          prefix,
-          ld["actor_id"],
-          ld["contract_id"],
-          ld["link_name"],
-          ld["provider_id"],
-          ld["values"] || %{}
-        )
-
-        {:reply, success_ack()}
-      else
-        _ ->
-          {:reply, failure_ack("Invalid link definition put request")}
-      end
-    end
-  end
-
-  # Remove a link definition
-  # This will first remove the link definition from memory, then publish the removal
-  # message to the stream, then publish the removal directly to the relevant provider via the
-  # RPC channel
-  defp handle_request({"linkdefs", "del"}, body, _reply_to, prefix) do
-    Tracer.with_span "Handle Linkdef Del (ctl)", kind: :server do
-      with {:ok, ld} <- Jason.decode(body),
-           true <- has_values(ld, ["actor_id", "contract_id", "link_name"]) do
-        HostCore.Linkdefs.Manager.del_link_definition(
-          prefix,
-          ld["actor_id"],
-          ld["contract_id"],
-          ld["link_name"]
-        )
-
-        {:reply, success_ack()}
-      else
-        _ ->
-          {:reply, failure_ack("Invalid link definition removal request")}
-      end
-    end
+  # are emitted to the appropriate stream and cached.
+  # THESE ARE NOW DEPRECATED
+  # Eventually the host will stop subscribing to these topics
+
+  defp handle_request({"linkdefs", "put"}, _body, _reply_to, prefix) do
+    {:reply,
+     failure_ack(
+       "Putting linkdefs through control interface is no longer supported. Please write to the WCMDCACHE_#{prefix} bucket directly"
+     )}
```

Comment:
yes, because the alternative is the wash client will try and make a call to put link defs and it'll get "no responders". While deprecated, I'd rather return an error than "no responders". The lattice server doesn't know what js domain it should write to, so it's not possible for it to take the 'old' request and place it in the new cache.

Reply:
I was proposing an alternative that messages on this deprecated topic get proxied to the correct behavior (setting a value in the KV bucket). That way older clients still work. But I forgot that we need to know the JS domain now, so maintaining compatibility isn't possible 😞

Reply:
in theory we could pull the js domain from the "first host running in the OTP process under the lattice prefix x" and use that for bucket putting.